cdap-api/src/main/java/co/cask/cdap/internal/workflow/ProgramWorkflowAction.java

```diff
@@ -42,11 +50,16 @@
   private String programName;
   private Runnable programRunner;
   private SchedulableProgramType programType;
+  private Map<String, Dataset> datasets;
+  private TransactionSystemClient txClient;
 
-  public ProgramWorkflowAction(String name, String programName, SchedulableProgramType programType) {
+  public ProgramWorkflowAction(String name, String programName, SchedulableProgramType programType,
+                               @Nullable TransactionSystemClient txClient) {
```

Comment:
ProgramWorkflowAction is purely for running another program (MR / Spark), it shouldn't be used for running custom action. @sagarkapare That should be the case, right?